### PR TITLE
Applications Exposed when logged off

### PR DESCRIPTION
**Pull request by AntBush**
The crux of this issue is that the `applications` table was given guest select permissions, where that is not wanted/needed in this case. This PR is to rectify that. This PR still requires tests - and as such is not ready to be merged in

### CODE REVIEW

**Comment by matthieu-foucault** [comment]
@AntBush https://github.com/bcgov/cas-cif/blob/develop/schema/test/unit/tables/project_test.sql has some permissions test examples.

Something like this would work. You'll want to add tests for insert, update and delete as well, and for the `ccbc_auth_user` role.
`delete` should be forbidden for all roles, as we only want to allow soft-delete.
```sql
set role ccbc_guest;
select throws_like(
  $$
    select * from ccbc_public.applications
  $$,
  'permission denied%',
  'ccbc_guest cannot select rows from table applications'
);
```

**Comment by VB-at-Bis** [comment]
Dashboard link will not be visible for non-logged in users with  https://app.zenhub.com/workspaces/ccbc---delivery-board-61c0c1204c8bcb001491c5f3/issues/bcgov/conn-ccbc-portal/294 . Direct access to /dashboard URL as a bookmark or browser link redirects user to Home page. Therefore, there are no steps in UI that can lead to the issue of `Applications Exposed when logged off`

**Comment by matthieu-foucault** [comment]
Although the UI doesn't expose applications for the `ccbc_guest` we still want to ensure that database permissions are correct and tested, so this change is still needed

**Comment by VB-at-Bis** [comment]
Tests added and verified locally:
```vb@vbpro:/mnt/c/Workspace/Button/CONN-CCBC-portal/db/test/unit/tables$ pg_prove -d ccbc ccbc_user_test.sql
ccbc_user_test.sql .. ok
All tests successful.
Files=1, Tests=21,  0 wallclock secs ( 0.03 usr  0.00 sys +  0.05 cusr  0.01 csys =  0.09 CPU)
Result: PASS
```

File: db/test/unit/tables/ccbc_user_test.sql [hunk 1]
```diff
@@ -116,5 +116,13 @@ select throws_like(
...
 );
 
+select throws_like(
+  $$
+    select * from ccbc_public.applications
+  $$,
+  'permission denied%',
+    'ccbc_guest cannot select rows from table_applications'
+);
+
```
Comment (matthieu-foucault) [inline db/test/unit/tables/ccbc_user_test.sql:126]: can we move this to a new `application_test.sql` file? 
We will want to test the other permissions as well (insert, update, delete)

**Review by matthieu-foucault — DISMISSED** [review]

File: db/test/unit/tables/ccbc_user_test.sql (hunk 1 above)
Comment (VB-at-Bis) [reply db/test/unit/tables/ccbc_user_test.sql]: Moved tests to `ccbc_guest_test.sql`

File: db/test/unit/tables/ccbc_guest_test.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
+begin;
```
Comment (marcellmueller) [inline db/test/unit/tables/ccbc_guest_test.sql:1]: I believe this should all stay in the same file. This test is in the `/tables` folder which was named after `ccbc_user` table.

It is modelled after this one but ours needs work: 
https://github.com/bcgov/cas-cif/blob/develop/schema/test/unit/tables/cif_user_test.sql

This morning in our meeting Alec asked us to rename ours for consistency here:
https://github.com/bcgov/CONN-CCBC-portal/pull/429/files

Though ours is a work in progress.

**Review by marcellmueller — APPROVED** [review]

File: db/test/unit/tables/ccbc_guest_test.sql (hunk 2 above)
Comment (VB-at-Bis) [reply db/test/unit/tables/ccbc_guest_test.sql:1]: Ok, lets merge this as is and re-factor tests in another ticket :)

**Review by marcellmueller — APPROVED** [review]

**Pull request merged**